modules/gdscript/gdscript_vm.cpp

```diff
@@ -1231,6 +1231,13 @@ Variant GDScriptFunction::call(GDScriptInstance *p_instance, const Variant **p_a
 
 				GD_ERR_BREAK(to_type < 0 || to_type >= Variant::VARIANT_MAX);
 
+#ifdef DEBUG_ENABLED
+				if (EngineDebugger::is_active() && src->get_type() == Variant::OBJECT && !src->operator ObjectID().is_ref_counted() && ObjectDB::get_instance(src->operator ObjectID()) == nullptr) {
```

Comment:
Is there any reason to check `EngineDebugger::is_active()`? Shouldn't this be checked in any debug build, even if not actively debugging? In that case the user can still see an error being reported in the output.

Reply:
I did it that way to match the similar checks present in `Variant`. However, I can see that other entities (like `Callable`) perform such checks regardless the debugger being active or not. So, the only real clue about keeping or removing that part of the condition will be what you tell me you want for GDScript.

Reply:
I'd say to not check for the debugger. It's somewhat common to test without a debugger attached, especially on mobile, so it's better to see a full error message in the log when it happens instead of when it hits something else.

Reply:
Done! (Also to the counterpart PR for 3.x.)